src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -510,6 +510,7 @@ private void HandleFinished(bool success, ClientSideStatus receivedStatus)
 
             lock (myLock)
             {
+                readingDone = true;
```

Comment:
The problem is the following:

1. we start a streaming call and immediately request the status (we need to do that to get notifications about the call being aborted in case there's a network error).
2a. let's assume at some point we received the status OK. At this point we can't really set readingDone to true and dispose the call, because there might be more messages to be delivered (also in general case we might have received the status on a different thread than where we receive the message) and we cannot risk throwing away messages for calls that ended with an OK status. Also, gRPC C# itself doesn't request the reads, the user does. So without the user actually reading the response stream to the end, we cannot really do much - what we would need to do is to somehow take over requesting the reads and make sure there are no more messages to be read.